pkg/controller/resourceclaim/controller.go

```diff
@@ -864,32 +862,27 @@ func (ec *Controller) syncClaim(ctx context.Context, namespace, name string) err
 		// for such claims and not checking for them keeps this code simpler.
 		if len(valid) == 0 {
 			if builtinControllerFinalizer >= 0 {
-				if claim.Spec.AllocationMode == resourcev1alpha2.AllocationModeWaitForFirstConsumer ||
-					claim.DeletionTimestamp != nil {
-					// Allocated by scheduler with structured parameters. We can "deallocate"
-					// by clearing the allocation.
-					claim.Status.Allocation = nil
-				}
-			} else if claim.Spec.AllocationMode == resourcev1alpha2.AllocationModeWaitForFirstConsumer {
+				// Allocated by scheduler with structured parameters. We can "deallocate"
+				// by clearing the allocation.
+				claim.Status.Allocation = nil
+			} else {
```

Comment:
This lost the check for `claim.DeletionTimestamp != nil`-- just want to make sure that was intended.

Reply:
Yes. `claim.Spec.AllocationMode == resourcev1alpha2.AllocationModeWaitForFirstConsumer` is always "true" now, so deallocation happens regardless whether the claim is deleted or not.

That's the behavior that we want.